django/forms/forms.py

```diff
@@ -580,7 +580,13 @@ def value(self):
         if not self.form.is_bound:
             data = self.form.initial.get(self.name, self.field.initial)
             if callable(data):
-                data = data()
+                # Get/set the result of a prior call from/to an attribute, to
+                # avoid having to call the initial callable twice (#24391);
+                # default to the result of calling it
```

Comment:
This comment seems a bit wordy. Could we simply say something like `Callables should be evaluated once per field.`? Maybe `self._initial_value` would be a slightly better variable name.